lua/packer/log.lua

```diff
@@ -150,4 +150,10 @@ end
 log.new(default_config, true)
 -- }}}
 
+function log.open()
+  -- This should be consistent with the format mentioned in log.new above.
+  local logfile = string.format('%s/%s.log', vim.fn.stdpath('cache'), default_config.plugin)
+  vim.cmd(string.format("tabedit + %s", logfile))
```

Comment:
Should we make the command for opening the log configurable? Maybe this should be tied to `config.display.open_fn`/`config.display.open_cmd`?

Reply:
One thought regarding this is that though this would allow floating windows to work it means that the user would have to have their packer update buffer etc. work the same as their log buffer so if you used a tab for logs you'd also get a tab for a status buffer but that isn't really blocking I think it means people will have to stick with splits for logs

Reply:
I'm sure there will be people who disagree (somewhat invalidating this comment 😆 ) but I think opening logs doesn't really make sense inside of a floating window, or a small split.

I appreciate that we're very much in the middle of the floating window trend but surely for readability of log files there isn't that much value to having them open in a bunch of different ways.

Anyway since people are likely to ask for flexibility regardless 🤷🏾

Reply:
No disrespect to people's preferences, but I don't think that should be configurable. The options to open are:
- Split: With this we would have 3 active splits in the current tabpage namely the current buffer, packer window and the log file. This would make viewing the logs uncomfortable with all the scrolling required.
- Floating window: In cases where the user wants to go back to either the packer window or the working buffer, the floating window will have to be closed and reopened again to view the logs.
- Tab: With the log file opened in a new tab, the user has the flexibility to move back and forth and closing the log file is just a single `q` away. It would also be comfortable to view the logs.

Similar examples: `:checkhealth`, `:LspInfo`
Although, the latter opens in a floating window.

---
This is not related to this PR but it seems that floating windows cannot be split further. This means if the user configured packer to be opened in a floating window and if they're trying to access the diff of a commit, it will open in a split in the previous buffer and the floating window will be on top of it. Thus, to access the diff window the user will have to close the floating window, but that would lose access to other diffs which the user would like to view.

Reply:
Alternatively, making it a user command (e.g. `:PackerLog`) would allow supporting modifiers. Split by default, `:vertical PackerLog` to split vertically, `:tab PackerLog` to open a tab (IIRC Neovim is supposed to get a `:float` modifier at some point as well).

It's less configuration for users and it mimics existing Ex commands

Reply:
Thanks for the discussion; these are reasonable points. I think I favor @nanotee's suggestion to allow supporting modifiers via the `PackerLog` command, *but* to have the keybinding in the results window bound to `:tab PackerLog`, to open in a tab if the results split/float is already open (incidentally, I'm unsure how opening a new tab works when you're in a floating window - I assume it's as normal, but there might be an undesired interaction there?)

Reply:
That seems like a good solution. So, correct me if I'm wrong, but:
- `PackerLog` should just open the log file in the _current_ buffer to make the modifiers to work correctly.
- The default behavior in the _packer_ window should be to open in a new tab.